### PR TITLE
Add CLI command to add supplemental dimensions

### PR DESCRIPTION
**Pull request by daniel-thom**
Closes GitHub Issue #304

### CODE REVIEW

File: dsgrid/cli/registry.py [hunk 1]
```diff
@@ -686,6 +686,36 @@ def update_project(
...
+@click.pass_obj
+@click.pass_context
+@click.argument("project_id")
+@click.argument("filename", callback=_path_callback)
+@click.option(
+    "-l",
+    "--log-message",
+    required=True,
+    type=str,
+    help="Reason for addition",
```
Comment (elainethale) [inline dsgrid/cli/registry.py:699]:
```suggestion
    help="Please specify the reason for this addition in your log message",
```

File: dsgrid/cli/registry.py [hunk 2]
```diff
@@ -686,6 +686,36 @@ def update_project(
...
+)
+def register_supplemental_dimensions(
+    ctx, registry_manager, project_id, filename: Path, log_message
+):
+    """Register new supplemental dimensions to a project."""
```
Comment (elainethale) [inline dsgrid/cli/registry.py:704]:
```suggestion
    """Register new supplemental dimensions with a project."""
```

File: dsgrid/config/supplemental_dimension.py [hunk 3]
```diff
@@ -19,3 +20,23 @@ class SupplementalDimensionModel(DimensionModel):
...
+
+class SupplementalDimensionsListModel(DSGBaseModel):
+    """Defines a list of supplemental dimensions."""
+
+    supplemental_dimensions: Annotated[
+        list[SupplementalDimensionModel],
+        Field(
+            title="supplemental_dimensions",
+            description="List of supplemental dimensions. They will be automatically registered "
+            "during project registration and then converted to supplemental_dimension_references.",
```
Comment (elainethale) [inline dsgrid/config/supplemental_dimension.py:33]: Is this model only used for the new CLI functionality to add supplemental dimensions to an already-registered project? If so, should this description say something different than "They will be automatically registered during project registration"?

File: dsgrid/cli/registry.py (hunk 1 above)
Comment (elainethale) [reply dsgrid/cli/registry.py:699]: Probably my suggested language doesn't fit with the rest of the CLI. How about, "Reason for adding these supplemental dimensions to the project"?

File: dsgrid/config/supplemental_dimension.py [hunk 4]
```diff
@@ -19,3 +20,12 @@ class SupplementalDimensionModel(DimensionModel):
             title="mapping",
         ),
     ]
+
+
+class SupplementalDimensionsListModel(DSGBaseModel):
+    """Defines a list of supplemental dimensions."""
+
+    supplemental_dimensions: Annotated[
```
Comment (lixiliu) [inline dsgrid/config/supplemental_dimension.py:28]: Is this true, a container for both dimensions and mappings? If so, can we name it as such?

Comment (daniel-thom) [reply dsgrid/config/supplemental_dimension.py:28]: This is directly analogous to this field: https://github.com/dsgrid/dsgrid/blob/main/dsgrid/config/project_config.py#L262. Would you prefer to call this `supplemental_dimensions_and_mappings` and lose the symmetry? Also, I could make an argument that a mapping is an attribute of a supplemental dimension and is therefore covered under this simpler name.

Comment (lixiliu) [reply dsgrid/config/supplemental_dimension.py:28]: If this is already in place in other config types, I am fine with keeping it as is. At one point, I think we have dimensions and mappings separate? So in my mental model, they are two distinct entities. Maybe it's time for me to update that.

Comment (daniel-thom) [reply dsgrid/config/supplemental_dimension.py:28]: We store them separately in the registry. However, at some point we changed the project config definition to allow them to be specified in one object, as in this example: https://github.com/dsgrid/dsgrid-project-DECARB/blob/main/project/project.json5#L320.

**Review by lixiliu — APPROVED** [review]

**Pull request merged**